parse/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+#[derive(Debug)]
+pub struct ParseFloatError {
+    kind: FloatErrorKind,
+}
+
+#[derive(Debug)]
+enum FloatErrorKind {
+    Empty,
+    Invalid,
+}
+
+fn pfe_empty() -> ParseFloatError {
+    ParseFloatError { kind: FloatErrorKind::Empty }
+}
+
+fn pfe_invalid() -> ParseFloatError {
+    ParseFloatError { kind: FloatErrorKind::Invalid }
+}
+
+pub fn parse_f64(string: &str) -> Result<f64, ParseFloatError> {
```

Comment:
I was going for compatability with the original parse routine. In rust, FloatErrorKind is private and ParseFloatError is the thing that can implement the Error trait, etc. I haven't done that yet, since it wasn't needed.